### ISSUE TITLE
Build error in RN 0.62 for iOS

### ISSUE THREAD

**Issue by erickjth** [rollbar/rollbar-react-native#108]
The iOS build is failing due to some compilation errors:

```
/AwesomeTSProject/ios/Pods/Rollbar/KSCrash/Source/KSCrash/Recording/Tools/KSDemangle_Swift.cpp:33:5: error: use of undeclared identifier 'swift'
    swift::Demangle::DemangleOptions options = swift::Demangle::DemangleOptions::SimplifiedUIDemangleOptions();
    ^
/AwesomeTSProject/ios/Pods/Rollbar/KSCrash/Source/KSCrash/Recording/Tools/KSDemangle_Swift.cpp:33:48: error: use of undeclared identifier 'swift'
    swift::Demangle::DemangleOptions options = swift::Demangle::DemangleOptions::SimplifiedUIDemangleOptions();
                                               ^
/AwesomeTSProject/ios/Pods/Rollbar/KSCrash/Source/KSCrash/Recording/Tools/KSDemangle_Swift.cpp:34:10: error: no type named 'string' in namespace 'std'
    std::string demangled = swift::Demangle::demangleSymbolAsString(mangledSymbol, options);
    ~~~~~^
AwesomeTSProject/ios/Pods/Rollbar/KSCrash/Source/KSCrash/Recording/Tools/KSDemangle_Swift.cpp:34:29: error: use of undeclared identifier 'swift'
    std::string demangled = swift::Demangle::demangleSymbolAsString(mangledSymbol, options);
                            ^
4 errors generated.
```
It is happening in a fresh installation.

My environment:
react-native-cli: 2.0.1
react-native: 0.62.0
cocoapod: 1.8.4

**Comment by taltrui** [comment]
Having the same issue, I've tried some workarounds like the target remove but with no luck. Did you find any solution?

**Comment by erickjth** [comment]
> Having the same issue, I've tried some workarounds like the target remove but with no luck. Did you find any solution?

No luck here, I tried a couple of things but nothing gets it working.

**Comment by taltrui** [comment]
Do you have Flipper enabled? I removed Rollbar to test, and a similar error happened on another library (react-native-material-kit). After removing Flipper the error went away. But since I do want to use Flipper it's not an optimal solution. 

I think the problem is that libraries are accessing Flipper-Folly and not Folly itself and that causes missing dependencies errors, but I've no clue on how to fix it, i'm trying tho, so if something comes up i'll update here.

**Comment by osikes** [comment]
same issue

**Comment by mlazari** [comment]
Not sure if it is the same issue, but I also see an error with a new RN project with react-native 0.62.2 after installing rollbar-react-native 0.7.2:
```
In file included from /Users/zzz/Desktop/test/AwesomeApp/ios/Pods/Rollbar/KSCrash/Source/KSCrash/Recording/Tools/KSDemangle_Swift.cpp:28:
/Users/zzz/Desktop/test/AwesomeApp/ios/Pods/Flipper-Folly/folly/Demangle.h:19:10: fatal error: 'folly/FBString.h' file not found
#include <folly/FBString.h>
```
I opened KSDemangle_Swift.cpp:28 and there is a `#include "Demangle.h"` directive. I think it tries to import `/Users/zzz/Desktop/test/AwesomeApp/ios/Pods/Rollbar/KSCrash/Source/KSCrash/swift/Basic/Demangle.h` but instead it imports `/Users/zzz/Desktop/test/AwesomeApp/ios/Pods/Flipper-Folly/folly/Demangle.h`.

**Comment by mlazari** [comment]
I managed to solve my error by adding this to `ios/Podfile`:

```
# Post Install processing for Rollbar
def rollbar_post_install(installer)
  installer.pods_project.targets.each do |target|
    if target.name == 'Rollbar'
      target.build_configurations.each do |config|
        config.build_settings['USE_HEADERMAP'] = 'NO'
      end
    end
  end
end

post_install do |installer|
    rollbar_post_install(installer)
end
```

**Comment by taltrui** [comment]
> I managed to solve my error by adding this to `ios/Podfile`:
> 
> ```
> # Post Install processing for Rollbar
> def rollbar_post_install(installer)
>   installer.pods_project.targets.each do |target|
>     if target.name == 'Rollbar'
>       target.build_configurations.each do |config|
>         config.build_settings['USE_HEADERMAP'] = 'NO'
>       end
>     end
>   end
> end
> 
> post_install do |installer|
>     rollbar_post_install(installer)
> end
> ```

That actually worked @mlazari! You are awesome 😄.

Mind telling me how you got to that? I'm not really versed in cocoapods so it it would be nice to know how to debug and know what is causing the error.

Thanks again!

**Comment by mlazari** [comment]
@taltrui 
1. First I looked at the error message and opened `ios/Pods/Rollbar/KSCrash/Source/KSCrash/Recording/Tools/KSDemangle_Swift.cpp` at line 28, and saw the `#include "Demangle.h"` directive.
2. I looked in Rollbar group under `Pods` group in Xcode and saw that there is a "Demangle.h" file that is not the same file as the file mentioned in the error.
3. I searched for this issue of including wrong file when there are multiple header files with the same name and found this answer: https://stackoverflow.com/a/4815982/3337236
4. Tried to change USE_HEADERMAP = NO in Pods project, Rollbar target and saw that it fixes the build error.
5. Tried to find a way to change that, so that it doesn't revert on next `pod install` and saw the `flipper_post_install` added by react-native in Pods file which was setting `SWIFT_VERSION` build setting for `YogaKit` target. I basically copied that function for changing `USE_HEADERMAP` setting in Rollbar target. Re-run `pod install` and saw that it works.

**Comment by taltrui** [comment]
@mlazari Thank you!

**Comment by daxfrost** [comment]
@waltjones looking forward to this getting fixed! This is going to affect everyone

**Comment by waltjones** [comment]
There's an open PR now in rollbar-ios that needs approval and release from the maintainer before it can be pulled in to rollbar-react-native. https://github.com/rollbar/rollbar-ios/pull/275

**Comment by ggborcean** [comment]
Internal: https://app.intercom.com/a/apps/rlisg82q/inbox/inbox/849049/conversations/5155344606

**Comment by waltjones** [comment]
Fix released in v0.8.0. https://github.com/rollbar/rollbar-react-native/releases/tag/v0.8.0
